qemu/tests/balloon_check.py

```diff
@@ -31,7 +31,7 @@ def __init__(self, test, params, env):
             sleep_time = 90
         logging.info("Waiting %d seconds for guest's applications up" % sleep_time)
         time.sleep(sleep_time)
-        self.env["balloon_test_setup_ready"] = True
+        self.params.setdefault("balloon_test_setup_ready", True)
```

Comment:
Should use `params["balloon_test_setup_ready"] = True` instead, btw, you forgot to change line 27.